darwin/utils.py

```diff
@@ -210,13 +280,15 @@ def find_files(
     for f in files:
         path = Path(f)
         if path.is_dir():
-            found_files.extend([f for f in path.glob(pattern) if is_extension_allowed("".join(f.suffixes))])
-        elif is_extension_allowed("".join(path.suffixes)):
+            found_files.extend([f for f in path.glob(pattern) if is_extension_allowed(str(path))])
```

Comment:
These changes made only by formatter

Reply:
Just to give visibility this line - `"".join(path.suffixes)`  is there to handle the fact that some extensions have two dots in them e.g `.nii.gz`. I'm not sure this new logic works, as `is_extension_allowed` expects only the extension passed to it not the entire path.

Reply:
Ah I see the new `is_extension_allowed_by_filename` but, maybe that should be used instead of `is_extension_allowed`

Reply:
Yeah, you're absolutely right, not sure how I missed that!

Well, I am - I _unit_ tested it, so tested the functions individually!